servicemodules/discord.py

```diff
@@ -99,7 +102,16 @@ async def on_message(message: discord.message.Message) -> None:
             responses = [Response()]
             for module in self.modules:
                 log.info(class_name, msg=f"# Asking module: {module}")
-                response = module.process_message(message)
+                try:
+                    response = module.process_message(message)
+                except Exception as e:
```

Comment:
Can you use a more narrow exception? What exact kind of error does module.process_message throw that you want to catch?

Reply:
They're not _supposed_ to throw anything except maybe `SystemExit` in the case of `s, reboot` that's why it needs to catch everything under `Exception` to report any bugs to the bot devs.  The only other alternative is to have Rob SSH into the bot to check logs, which isn't ideal.  Without this, there is a `try: except Exception:` wrapping `on_message` in the discord py backend dropping the error.  This try catch makes the error handling actually useful, and allows the message to be responded to even if one module fails. (Currently any failure prevents a response)